### PR TITLE
fix: include orgs slug in getUserByUsername just like [user]/[type] page

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Handles ORG when looking up userId by username. This is the same way we do it on the SSR of booking pages.

Fixes # (issue)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Are there environment variables that should be set? No.
- What are the minimal test data to have? An Org with a username that it's the same as other without organization.
- What is expected (happy path) to have (input and output)? They should have their unique availability on their page if they both have the same eventSlug.
- Any other important info that could help to test that PR

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/trpc/server/routers/viewer/slots/util.ts [hunk 1]
```diff
@@ -211,20 +215,28 @@ export async function getDynamicEventType(input: TGetScheduleInputSchema) {
...
-  return isDynamicBooking ? getDynamicEventType(input) : getEventType(input);
+  return isDynamicBooking ? getDynamicEventType(input) : getEventType(input, organizationDetails);
 }
 
-export async function getAvailableSlots(input: TGetScheduleInputSchema) {
+export async function getAvailableSlots(
+  ctx: { req: { headers: { host?: string | null } } },
+  input: TGetScheduleInputSchema
+) {
+  const orgDetails = orgDomainConfig(ctx.req.headers.host ?? "");
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/slots/util.ts:230]: Getting org slug from contex just as we do on booking pages.

File: packages/trpc/server/routers/viewer/slots/util.ts [hunk 2]
```diff
@@ -480,10 +492,15 @@ export async function getAvailableSlots(input: TGetScheduleInputSchema) {
...
-async function getUserIdFromUsername(username: string) {
+async function getUserIdFromUsername(
+  username: string,
+  organizationDetails: { currentOrgDomain: string | null; isValidOrgDomain: boolean }
+) {
+  const { currentOrgDomain, isValidOrgDomain } = organizationDetails;
   const user = await prisma.user.findFirst({
     where: {
       username,
+      organization: isValidOrgDomain && currentOrgDomain ? getSlugOrRequestedSlug(currentOrgDomain) : null,
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/slots/util.ts:503]: This was required if an org user have the same username with another one without org.

**Comment by zomars** [comment]
We got a type error 
image

**Review by zomars — CHANGES_REQUESTED** [review]
Let's fix the types first 🙏🏽

**Comment by alannnc** [comment]
@zomars No more type erros on this :)

**Review by Udit-takkar — DISMISSED** [review]
Getting 404 on getSchedule

If a user creates a team "engineering" and another user has a team name "engineering" but in his organisation then i get 404 on getSchedule when i go the booking page.

Screenshot 2023-08-23 at 12 27 22 PM

**Comment by alannnc** [comment]
> Getting 404 on getSchedule
> 
> If a user creates a team "engineering" and another user has a team name "engineering" but in his organisation then i get 404 on getSchedule when i go the booking page.
> 
> Screenshot 2023-08-23 at 12 27 22 PM

Nice find @Udit-takkar, it was a different bug from what I was solving but related as I applied the same logic to solve team names collition.

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**